src/kotlin/kage/format/AgeHeader.kt

```
@@ -76,37 +78,37 @@ public class AgeHeader(public val recipients: List<AgeStanza>, public val mac: B
      * Age Spec:
      * The first line of the header is age-encryption.org/ followed by an arbitrary version string.
      */
-    internal fun parseVersionLine(versionLine: String) {
+    private fun parseVersionLine(versionLine: String?) {
       if (versionLine != VERSION_LINE)
         throw InvalidVersionException("Version line is not correct: $versionLine")
     }
 
-    internal fun parseRecipients(reader: BufferedReader): List<AgeStanza> {
+    internal fun parseRecipients(reader: BufferedInputStream): List<AgeStanza> {
       val recipientList = mutableListOf<AgeStanza>()
-      val characterArray = CharArray(3)
+      val buf = ByteArray(3)
 
       while (true) {
         // Add a mark to be able to reset the reader after reading the first 3 characters of the
         // line
         reader.mark(3)
-        if (reader.read(characterArray) == -1)
+        if (reader.read(buf) == -1)
           throw InvalidRecipientException("End of stream reached while reading recipients")
 
-        val line = characterArray.concatToString()
+        val prefix = buf.decodeToString()
         reader.reset()
 
-        if (line.startsWith(RECIPIENT_PREFIX)) {
+        if (prefix.startsWith(RECIPIENT_PREFIX)) {
           recipientList.add(AgeStanza.parse(reader))
-        } else if (line.startsWith(FOOTER_PREFIX)) {
+        } else if (prefix.startsWith(FOOTER_PREFIX)) {
           return recipientList
         } else {
           throw InvalidRecipientException("Unexpected line found: ${reader.readLine()}")
         }
       }
     }
 
-    internal fun parseFooter(reader: BufferedReader): ByteArray {
-      val footerLine = reader.readLine()
+    internal fun parseFooter(reader: BufferedInputStream): ByteArray {
+      val footerLine = reader.readLine() ?: ""
```

I think we should throw an exception here if the `footerLine` is not available